### ISSUE TITLE
[BUG] TextEmbeddingProcessor has kind "inference"

### ISSUE THREAD

**Issue by maxlepikhin** [opensearch-project/opensearch-java#1149]
### What is the bug?
TextEmbeddingProcessor has kind "inference".

### How can one reproduce the bug?
See [code](https://github.com/opensearch-project/opensearch-java/blob/08e7e6504d4e6029640940f4bb4670e5e183c700/java-client/src/main/java/org/opensearch/client/opensearch/ingest/TextEmbeddingProcessor.java#L54).

### What is the expected behavior?
[Docs](https://opensearch.org/docs/latest/ingest-pipelines/processors/text-embedding/) uses "text_embedding" kind.

### What is your host/environment?
Ubuntu 22.04

### Do you have any screenshots?
N/A

### Do you have any additional context?
N/A

**Comment by Xtansia** [comment]
There's an open PR #1127 that will fix this.

Using `new Processor.Builder().textEmbedding(...)` will set the kind correctly:
```java
            client.ingest()
                .putPipeline(
                    new PutPipelineRequest.Builder().id(INGEST_PIPELINE_NAME)
                        .description("A test_embedding ingest pipeline for the opensearch-java " + SAMPLE_NAME + " sample")
                        .processors(
                            new Processor.Builder().textEmbedding(
                                new TextEmbeddingProcessor.Builder().modelId(modelId).fieldMap("text", "passageEmbedding").build()
                            ).build()
                        )
                        .build()
                );
```

This issue only affects construction via `new Processor(textEmbeddingProcessor)`.

**Comment by Xtansia** [comment]
The fix for this was released in version 2.15.0
